ql/src/java/org/apache/hadoop/hive/ql/io/CombineHiveInputFormat.java

```diff
@@ -370,7 +371,8 @@ private InputSplit[] getCombineSplits(JobConf job, int numSplits,
       PartitionDesc part = HiveFileFormatUtils.getFromPathRecursively(
           pathToPartitionInfo, path, IOPrepareCache.get().allocatePartitionDescMap());
       TableDesc tableDesc = part.getTableDesc();
-      if ((tableDesc != null) && tableDesc.isNonNative()) {
+      boolean useDefaultFileFormat = part.getInputFileFormatClass().isAssignableFrom(tableDesc.getInputFileFormatClass());
+      if (tableDesc != null && tableDesc.isNonNative() && useDefaultFileFormat) {
```

Comment:
tableDesc can't be null here, 1 line above you use it without any checks

Reply:
Added tableDesc != null to cover both scenarios.